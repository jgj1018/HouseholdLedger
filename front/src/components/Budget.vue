<template>
  <div class="hello">
    <form id="loginForm" @submit.prevent="createBudget" method="post">
      <input type="text" v-model="budgetName" placeholder="budgetName">
      <select v-model="type">
          <option value=""></option>
          <option :value="item.code"
          v-for="(item, idx) in budgetType" :key="idx"
          >{{item.name}}</option>
      </select>
      <input type="number" v-model="costAmount" placeholder="amount">
      <div>
          <ul>
            <li v-bind:key="param" v-for="(messages, param) in errors">
              {{param}} {{messages}}
            </li>
          </ul>
      </div>
      <button type="submit" >Register</button>
      <ul>
        <li v-for="(item, idx) in budget" :key="idx">
          {{idx}} : {{item}}
        </li>
      </ul>
    </form>
  </div>
</template>

<script>
import Http from '../config/Http'
import Api from '../config/Api'
import Budget from '../vo/BUdget'
import Charts from './Profit'

export default {
  name: 'Budget',
  data () {
    return {
      budgetType: [],
      type: null,
      budgetName: '',
      costAmount: null,
      budget: [],
      errors: []

    }
  },
  methods: {
    createBudget: async function (e) {
      let budgetObj = new Budget(this.budgetName, this.type, this.costAmount)
      let budgetVue = this

      Http.post(Api.budget.create, budgetObj).then(async function (r) {
        if (r.status === 201) {
          let budget = await Http.get(Api.budget.list)
          console.log('BUDGET', budget)
          budgetVue.budget = budget.data
        }
      }).catch(function (error) {
        console.dir(error.response.data)
        this.errors = error.response.data
      })
    }
  },
  created: async function () {
    let data = await Http.get(Api.bootUp.getBudgetType)
    this.budgetType = data.data
    let budgetList = await Http.get(Api.budget.list)
    console.dir(budgetList)
    this.budget = budgetList.data
  },
  components: {
    Charts
  }

}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>

</style>
