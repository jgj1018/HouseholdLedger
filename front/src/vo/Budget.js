
export default class Budget {
  constructor (budgetName, budgetType, costAmount) {
    this.budget_name = budgetName
    this.budget_type = budgetType
    this.cost_amount = costAmount
  }
}
