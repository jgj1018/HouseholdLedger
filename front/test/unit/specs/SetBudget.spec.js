import { shallow } from 'vue-test-utils'
import Budget from '@/components/Budget'

describe('Set Budget Test.vue', () => {
  let cmp = null
  cmp = shallow(Budget)
  describe('Budget.vue Data Test', () => {
    it('structure must be same', () => {
      expect(cmp.element).toMatchSnapshot()
    })
  })
}
)
