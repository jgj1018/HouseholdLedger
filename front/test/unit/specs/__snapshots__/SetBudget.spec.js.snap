// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Set Budget Test.vue Budget.vue Data Test structure must be same 1`] = `
<div
  class="hello"
>
  <form
    id="loginForm"
    method="post"
  >
    <select>
      <option
        value=""
      />
       
    </select>
     
    <input
      placeholder="amount"
      type="number"
    />
     
    <div>
      <ul />
    </div>
     
    <button
      type="submit"
    >
      Register
    </button>
  </form>
</div>
`;
